inc/type-post.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Posts Sitemap (for post-type posts).
+ */
+
+defined( 'ABSPATH' ) || die();
+
+// Register the a sitemap for the post post-type.
+add_filter( 'core_sitemaps_register_post_types', static function ( $post_types ) {
+	$post_types['post'] = 'core_sitemaps_type_post_register';
```

Comment:
This is an ok to get an initial idea, but I think that eventually an object oriented approach that uses a simplified [Registry pattern](https://designpatternsphp.readthedocs.io/en/latest/Structural/Registry/README.html) could be really useful here so we could register different Sitemap classes for each object type. [Yoast does something similar here](https://github.com/Yoast/wordpress-seo/blob/trunk/inc/sitemaps/class-sitemaps.php#L122-L142) which is a nice approach.